### ISSUE TITLE
Is there any plan to implement window function nth_value in the near future? We'd like to use it.

### ISSUE THREAD

**Issue by ryzuo** [ClickHouse/ClickHouse#26112]
In some of our scenarios, use nth_value can be more convenient for developers, saving them from writing much more codes. Though, I have an implementation of this function in WindowTransform.cpp that satisfy ourselves' usage, I'm thinking it's better to ask community if there's plan to implement it. Or I can create a pull request.

**Comment by kitaisreal** [comment]
@akuzm do we plan to implement it ?

**Comment by akuzm** [comment]
`nth_value(value, n)` -- Returns value evaluated at the row that is the n'th row of the window frame (counting from 1);

I think it can be replaced with `groupArray(n)(value)[n]`, but it is somewhat ugly, yes. We don't have plans to implement it soon, but a PR would be welcome.

**Comment by ryzuo** [comment]
Yes, groupArray is an option we tried, but with some more complicated requirements, the query can be much more complicated, and does not work very well and neither graceful.
I will shoot a PR later

**Comment by alexey-milovidov** [comment]
@akuzm Is it somehow difficult to implement? Maybe it's a piece of cake. Just tell what parts of code needs to me modified :)

**Comment by akuzm** [comment]
> @akuzm Is it somehow difficult to implement? Maybe it's a piece of cake. Just tell what parts of code need to me modified :)

We have an endless supply of "piece of cake" tasks that can saturate several full-time employees for years, so I'm not very keen to implement everything that pops up.

To answer your question, it needs about one meaningful line of code, ten lines of boilerplate and one test query. Analogous to `first_value`. The issue author says they have an implementation, so probably they have figured it out already.

**Comment by alexey-milovidov** [comment]
@ryzuo please send a PR.

**Comment by akuzm** [comment]
Implemented in https://github.com/ClickHouse/ClickHouse/pull/26334
